Hi Phil, just testing before submitting.

pmatar28code opened the pull request:
still need to fix strings , but that's it for today, gonna test on my Phone

pmatar28code committed:
starting edit recipes

pmatar28code committed:
edit seems to be working. edit recipe and fragment edit recipe is wha ti worked on, and the onclickedit on recipes fragment / adapter

edit seems to be working. edit recipe and fragment edit recipe is wha ti worked on, and the onclickedit on recipes fragment / adapter

pmatar28code committed:
updating recycler/adapter for delete ingredients and adding recipe works

pmatar28code committed:
updating adapter/recycler when editing recipes works

pmatar28code committed:
updating adapter/recycler when deleting recipe works

pmatar28code committed:
updating recycler/adapter when copying recipe works

updating recycler/adapter when copying recipe works

pmatar28code committed:
cleaned up code in main

pmatar28code committed:
fixed layout categories fragment removed actionbar from themes

pmatar28code committed:
added images to category, fixed item_categories layout bigger text and imageview

added images to category, fixed item_categories layout bigger text and imageview. categories adapter now set images

pmatar28code committed:
label visibility bottom navigation main layout

pmatar28code committed:
fixed ingredients fragment layout

pmatar28code committed:
fixed layout recipes fragment

pmatar28code committed:
fixed item ingredients layout

pmatar28code committed:
changed delete ingredient fab icon and fixed item ingredients layout again

pmatar28code committed:
got image working for recipes, works on edit, copy and delete, tested on recipe details also, tested on app restart so database is saving it, remove default recipes list, it starts blank on recipes

pmatar28code committed:
fixed issue when editing recipe and changing an image without entering to select ingredients, ingredients would be deleted, solution similar to image problems i had

fixed issue when editing recipe and changing an image without entering to select ingredients, ingredients would be deleted, solution similar to image problems i had

pmatar28code committed:
item layout for recipes fixed

pmatar28code committed:
added onclicklistener to image in item recipes, and viewholder to take user to details fragment

added onclicklistener to image in item recipes, and viewholder to take user to details fragment

pmatar28code committed:
fixed recipedetails layout

pmatar28code committed:
added functionality to edit button on details fragment , sort of workaround

pmatar28code committed:
added back button on details layout, added functionality to copy and delete recipe buttons on details fragment

added back button on details layout, added functionality to copy and delete recipe buttons on details fragment

pmatar28code committed:
added scroll view to display all content correctly on details fragment layout, changed bottom constraint for buttons

pmatar28code committed:
added remove ingredient deleted from ingredients from all recipes, delete ingredient dialog was modified and changed to var ingredients to use parameter of recipes object

pmatar28code committed:
fixed search and delete ingredients dialog, added cancel to editrecipe and add recipe ingredients dialog

fixed search and delete ingredients dialog, added cancel to editrecipe and add recipe ingredients dialog

pmatar28code committed:
added default image to add and edit layouts also to details layout by getting uri of defaultimage

pmatar28code committed:
added defaultimage uri to recipes adapter crated a var in repository and a function in main to achieve it

pmatar28code committed:
added background image for all layouts, corrected add ingredient layout

pmatar28code committed:
added default recipe, fixed edit fragment to show default image, added constraint layout to fab in ingredients layout

pmatar28code committed:
ingredients fragment  fab constraint, center crop on images for add ,edit,details, item recipes

pmatar28code committed:
edited text size to large in dropdown menu for search,add recipe, edit recipe

pmatar28code committed:
edited text size to large in dropdown menu for search,add recipe, edit r…

pmatar28code committed:
added text to edit and add recipe for image selection

pmatar28code committed:
cleaned up code on all files, added onclick to itemview in categories adapter

pmatar28code committed:
changed default recipe to include larger instructions, added instructions as string

pmatar28code committed:
added scroll view to add and edit recipe layout

pcutting commented on app/src/main/java/com/example/foodmanchu/EditRecipeDialog.kt:
```diff
@@ -0,0 +1,166 @@
+        val recipe = Repository.recipeToEdit
+        val listOfIngredientsS = recipe.ingredientsToUse.split(",")
+        for(ingredient in Repository.IngredientsList){
+            if(listOfIngredientsS.contains(ingredient.ingredientName) || listOfIngredientsS.contains(ingredient.ingredientName+" ,") ||  listOfIngredientsS.contains(" "+ingredient.ingredientName)){
```
Sometimes it's good to break long complex conditions into a function.  There is a refactor tool that helps.  "function with scope" i think it's called.

pcutting commented on app/src/main/java/com/example/foodmanchu/EditRecipeDialog.kt:
```diff
@@ -0,0 +1,166 @@
+        }
+        ingredientsSelectedString = ingredientsSelectedString.dropLast(1)
+        val newRecipe = Recipes(
+                recipeName = binding.editRecipeNameEditText.text?.toString()?:"",
```
spacing

```suggestion
                recipeName = binding.editRecipeNameEditText.text?.toString() ?: "",
```

pcutting commented on app/src/main/java/com/example/foodmanchu/EditRecipeDialog.kt:
```diff
@@ -0,0 +1,166 @@
+        ingredientsSelectedString = ingredientsSelectedString.dropLast(1)
+        val newRecipe = Recipes(
+                recipeName = binding.editRecipeNameEditText.text?.toString()?:"",
+                ingredientsToUse = if(ingredientsSelectedString ==""){
```
spacing

```suggestion
                ingredientsToUse = if (ingredientsSelectedString == ""){
```

pcutting commented on app/src/main/java/com/example/foodmanchu/EditRecipeDialog.kt:
```diff
@@ -0,0 +1,166 @@
+                recipeName = binding.editRecipeNameEditText.text?.toString()?:"",
+                ingredientsToUse = if(ingredientsSelectedString ==""){
+                    binding.editTextView.text.toString()
+                }else{
```
spacing.  just some examples, but go through and clean it up.  maybe use find/replace to speed it up on some of the larger files.  

spacing is important because it helps the eye see the code in a regular way and helps you find your bugs, makes it easier to find logical bugs also.



```suggestion
                } else {
```

pcutting commented on app/src/main/java/com/example/foodmanchu/EditRecipeDialog.kt:
```diff
@@ -0,0 +1,166 @@
+        super.onActivityResult(requestCode, resultCode, intent)
+        if(resultCode == Activity.RESULT_OK && requestCode == PICK_IMAGE){
+            intent?.run{
+                imageUri = data ?: "".toUri()
```
I keep wondering about this.  something about it teases me. Don't touch it for now.

pcutting commented on app/src/main/java/com/example/foodmanchu/RecipeDetailsFragment.kt:
```diff
@@ -1,26 +1,76 @@
         val binding = RecipeDetailsFragmentBinding.bind(view)
         super.onViewCreated(view, savedInstanceState)
-
+        lateinit var mainActivity: MainActivity
```
I got a bad feeling about this one.

pcutting commented on app/src/main/java/com/example/foodmanchu/RecipeDetailsFragment.kt:
```diff
@@ -1,26 +1,76 @@
+        lateinit var mainActivity: MainActivity
         val recipe = Repository.listOfRecipeForDetail[0]
 
+        val defaultImageId = R.drawable.ic_default_image// r.mipmap.yourmipmap; R.drawable.yourdrawable
```
get rid of commented out code at this point.

pcutting commented on app/src/main/java/com/example/foodmanchu/RecipeDetailsFragment.kt:
```diff
@@ -1,26 +1,76 @@
+            }
+            detailsDuplicateRecipeButton.setOnClickListener {
+                var duplicateRecipe = Recipes(
+                        recipeName = recipe.recipeName +"_copy"+ Date().time.toString(),
```
use string interpolation, not concatenation.  

```suggestion
                        recipeName = "${recipe.recipeName} $_copy ${Date().time.toString()}",
```

pcutting commented on app/src/main/java/com/example/foodmanchu/Repository.kt:
```diff
@@ -4,8 +4,21 @@ import android.content.Context
 object Repository {
+    var repoDefaultImageUri ="".toUri()
+
+    var recipeToEdit = Recipes(
+            recipeName = "",
+            ingredientsToUse = "",
+            description = "",
+            cookingInstructions = "",
+            prepTime = "",
+            recipeCategory = "",
+            recipeImage = ""
+    )
```
if you are going to allow that, just consider making default parameters in the constructors.

pcutting commented on app/src/main/java/com/example/foodmanchu/Repository.kt:
```diff
@@ -165,12 +169,13 @@ object Repository {
+                    cookingInstructions = "To assemble, spread 1 1/2 cups of meat sauce in the bottom of a 9x13-inch baking dish. Arrange 6 noodles lengthwise over meat sauce. Spread with one half of the ricotta cheese mixture. Top with a third of mozzarella cheese slices. Spoon 1 1/2 cups meat sauce over mozzarella, and sprinkle with 1/4 cup Parmesan cheese. Repeat layers, and top with remaining ...
+                    prepTime = "45 minutes",
+                    recipeCategory = "Meat",
+                    recipeImage = ""
```
default parameters again :)-

pcutting commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -19,6 +19,7 @@
         android:id="@+id/bottomNavigationView"
         android:layout_width="match_parent"
         android:layout_height="0dp"
+        app:labelVisibilityMode="labeled"
         app:layout_constraintBottom_toBottomOf="parent"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
```
layout_ above other properties.
getting closer.  good.

```suggestion
        app:layout_constraintBottom_toBottomOf="parent"
        app:layout_constraintEnd_toEndOf="parent"
        app:layout_constraintStart_toStartOf="parent"
        app:labelVisibilityMode="labeled"
```

pcutting commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -19,6 +19,7 @@
         android:id="@+id/bottomNavigationView"
```
wrong format
```suggestion
        android:id="@+id/bottom_navigation_view"
```

pcutting commented on app/src/main/res/layout/categories_fragment.xml:
```diff
@@ -8,10 +8,27 @@
+        android:layout_marginEnd="@dimen/standard_margin"
+        android:text="Categories"
+        android:textAlignment="center"
+        android:textSize="24sp"
```
dimens file stuff

pcutting commented on app/src/main/res/layout/fragment_add_ingredient.xml:
```diff
@@ -1,35 +1,43 @@
+        android:layout_marginStart="@dimen/standard_margin"
+        android:layout_marginTop="@dimen/standard_margin"
+        android:layout_marginEnd="@dimen/standard_margin"
+        android:text="Add Ingredient"
```
@string !! mr bilingual.

pcutting commented on app/src/main/res/layout/fragment_add_ingredient.xml:
```diff
@@ -1,35 +1,43 @@
+        android:layout_marginEnd="@dimen/standard_margin"
+        android:text="Add Ingredient"
+        android:textAlignment="center"
+        android:textSize="24sp"
```
@dimens.

pcutting replied:
dimens, all over the place

pcutting commented on app/src/main/res/layout/fragment_add_ingredient.xml:
```diff
@@ -1,35 +1,43 @@
+        android:layout_marginStart="@dimen/standard_margin"
+        android:layout_marginTop="@dimen/standard_margin"
+        android:layout_marginEnd="@dimen/standard_margin"
+        android:hint="Ingredient Name"
```
@ string  .  that was probably a warning.

pcutting commented on app/src/main/res/layout/fragment_add_recipe.xml:
```diff
@@ -1,127 +1,169 @@
-    xmlns:android="http://schemas.android.com/apk/res/android"
+<ScrollView xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
```
if this is grayed out, you can get rid of it.

pcutting commented on app/src/main/res/layout/fragment_edit_recipe.xml:
```diff
@@ -0,0 +1,169 @@
+<ScrollView xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/scrollView"
```
name not descriptive. and "scroll_view" or just "scrollview".  watch mixing cases in id's.

pcutting commented on app/src/main/res/layout/fragment_edit_recipe.xml:
```diff
@@ -0,0 +1,169 @@
+
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
```
can you get rid of this? (xnls:tools...)

pcutting commented on app/src/main/res/layout/fragment_edit_recipe.xml:
```diff
@@ -0,0 +1,169 @@
+        android:layout_height="wrap_content"
+        android:layout_marginStart="@dimen/standard_margin"
+        android:layout_marginEnd="@dimen/standard_margin"
+        android:text="Edit Recipe"
```
@string

pcutting commented on app/src/main/res/layout/fragment_edit_recipe.xml:
```diff
@@ -0,0 +1,169 @@
+        android:layout_marginEnd="@dimen/standard_margin"
+        android:text="Edit Recipe"
+        android:textAlignment="center"
+        android:textSize="24sp"
```
@dimen

pcutting commented on app/src/main/res/layout/fragment_edit_recipe.xml:
```diff
@@ -0,0 +1,169 @@
+        android:text="Edit Recipe"
+        android:textAlignment="center"
+        android:textSize="24sp"
+        android:textStyle="bold"
```
This is a personal opinion:

If you are using things like bold, you are really saying that this is a view that i want to stand out.  Thus you should make a style for that, so that you can change them all at one time,  the designer may say that all of those should be in a different font and with other specific properties.  

Consider using styles.

pcutting commented on app/src/main/res/layout/fragment_edit_recipe.xml:
```diff
@@ -0,0 +1,169 @@
+        android:layout_height="wrap_content"
+        android:layout_marginStart="@dimen/standard_margin"
+        android:layout_marginEnd="@dimen/standard_margin"
+        android:hint="Recipe Description"
```
@string

pcutting commented on app/src/main/res/layout/fragment_edit_recipe.xml:
```diff
@@ -0,0 +1,169 @@
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/edit_recipe_prep_time_layout"
+        android:textAppearance="?android:attr/textAppearanceLarge">
```
out of order.

pcutting commented on app/src/main/res/layout/fragment_edit_recipe.xml:
```diff
@@ -0,0 +1,169 @@
+        app:layout_constraintTop_toBottomOf="@+id/edit_textView" />
+
+    <TextView
+        android:id="@+id/edit_textView"
```
watch case.

pcutting commented on app/src/main/res/layout/fragment_edit_recipe.xml:
```diff
@@ -0,0 +1,169 @@
+        tools:srcCompat="@tools:sample/backgrounds/scenic" />
+
+    <TextView
+        android:id="@+id/textView7"
```
use descriptive id's

pcutting commented on app/src/main/res/layout/fragment_search.xml:
```diff
@@ -3,15 +3,19 @@
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toTopOf="parent">
+        app:layout_constraintTop_toBottomOf="@+id/imageView">
```
non descriptive id, watch case

pcutting commented on app/src/main/res/layout/fragment_search.xml:
```diff
@@ -37,26 +46,41 @@
+        android:layout_marginTop="100dp"
+        android:layout_marginEnd="@dimen/standard_margin"
+        android:layout_marginBottom="@dimen/standard_margin"
+        android:text="Search"
```
@string

pcutting commented on app/src/main/res/layout/fragment_search.xml:
```diff
@@ -37,26 +46,41 @@
-        android:id="@+id/clear_button"
-        android:layout_width="wrap_content"
+    <TextView
+        android:id="@+id/textView5"
```
use descriptive id's.

pcutting commented on app/src/main/res/layout/fragment_search.xml:
```diff
@@ -37,26 +46,41 @@
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <ImageView
+        android:id="@+id/imageView"
```
use descriptive id's

pcutting commented on app/src/main/res/layout/ingredients_fragment.xml:
```diff
@@ -30,10 +34,25 @@
+        app:srcCompat="@drawable/ic_remove" />
+
+    <TextView
+        android:id="@+id/textView3"
```
use descriptive id's

pcutting commented on app/src/main/res/layout/item_categories.xml:
```diff
@@ -7,9 +7,29 @@
         android:id="@+id/categorie_text"
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
+        android:layout_width="166dp"
+        android:layout_height="35dp"
```
Just a comment: 

Use caution when specific width.  If you change the language/text you can destroy your view.

pcutting commented on app/src/main/res/layout/recipe_details_fragment.xml:
```diff
@@ -1,68 +1,166 @@
+<ScrollView xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/scrollView"
```
non descriptive id, casing.

pcutting commented on app/src/main/res/layout/recipe_details_fragment.xml:
```diff
@@ -1,68 +1,166 @@
+        android:layout_marginEnd="@dimen/standard_margin"
         android:text="TextView"
+        android:textAlignment="viewStart"
+        android:textSize="18sp"
+        android:textStyle="bold"
```
consider use a @style.

pcutting commented on app/src/main/res/layout/recipe_details_fragment.xml:
```diff
@@ -1,68 +1,166 @@
+        android:layout_marginEnd="@dimen/standard_margin"
+        android:scaleType="centerCrop"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="0.497"
```
0.497?????

pcutting commented on app/src/main/res/values/strings.xml:
```diff
@@ -1,4 +1,6 @@
 <resources>
     <string name="app_name">FoodManChu</string>
     <string name="fragment_delete_title_text">Please select ingredient to delete from the dropdown menu</string>
+    <string name="instructions_recipe">To assemble, spread 1 1/2 cups of meat sauce in the bottom of a 9x13-inch baking dish. Arrange 6 noodles lengthwise over meat sauce. Spread with one half of the ricotta cheese mixture. Top with a third of mozzarella cheese slices. Spoon 1 1/2 cups meat sauce over mozzarella, and sprinkle with 1/4 cup Parmesan cheese. Repeat layers, and top with remaining moz ...
+                    Bake in preheated oven for 25 minutes. Remove foil, and bake an additional 25 minutes. Cool for 15 minutes before serving.</string>
```
should this be in database? See if this is still used.

pcutting reviewed (commented):
Correct, update PR and Finish/Merge.

pmatar28code committed:
added default recipe and fix code layout in repository

pmatar28code committed:
corrected all warnings, added string resources to layouts, deleted an ingredients selection adapter i didnt need

corrected all warnings, added string resources to layouts, deleted an ingredients selection adapter i didnt need

pmatar28code committed:
more code cleanup

pmatar28code committed:
added images for default recipes via function to create uri and made it a string

pmatar28code committed:
added function to create uri on repository and assing uri to default recipes list , added context :context to companion object so i could access the fun requirements on repository, added hint to search filter dropdown menu

added function to create uri on repository and assing uri to default recipes list , added context :context to companion object so i could access the fun requirements on repository, added hint to search filter dropdown menu

pmatar28code committed:
added subtitle to recipes fragment layout, resource string

pmatar28code committed:
corrected layout on recipes fragment

pmatar28code committed:
added string resource and hint to fragment search layout

pmatar28code committed:
corrected adding new recipe, had past used ingredients. cleared repo list for ingredients used for recipe

pmatar28code committed:
corrected fab on recipes fragment, changed back button on details for fab
